ang/civicase/BulkActionsCheckboxes.js

```diff
@@ -0,0 +1,44 @@
+(function (angular, $, _) {
+  var module = angular.module('civicase');
+
+  module.directive('civicaseBulkActionsCheckboxes', function () {
+    return {
+      restrict: 'EA',
+      controller: 'civicaseBulkActionsController',
+      templateUrl: '~/civicase/BulkActionsCheckboxes.html',
+      scope: {
+        showCheckboxes: '=?',
+        selectedItems: '=',
+        isSelectAllAvailable: '='
+      }
+    };
+  });
+
+  module.controller('civicaseBulkActionsController', function ($scope, $rootScope) {
+    $scope.showCheckboxes = false;
+
+    $scope.$watch('selectedItems', function () {
+      $rootScope.$broadcast('stickyCaseListHeaderReinitiatePos');
```

Comment:
We can't use $emit as its subscribers will only be available inside its parent hierarchy, but where it is subscribers are added, are outside that scope.
So if we emit `stickyCaseListHeaderReinitiatePos` from `BulkActionsCheckboxes` it can't be caught in CaseListDirective as it doesn't come in `BulkActionsCheckboxes` parent Hierarchy.

Let me know if we need discussion on it.

Reply:
Yes, but if we always use `$rootScope.$watch` then this would work.